### PR TITLE
Add GSoC JSON files for 2020-2023 with custom tags

### LINKED ISSUE #167: Populate GSoC 2020-2023 OWASP projects with respective custom tags
This is a chore task to go through the OWASP site pages:
  - https://owasp.org/www-community/initiatives/gsoc/gsoc2020ideas
  - https://owasp.org/www-community/initiatives/gsoc/gsoc2021ideas
  - https://owasp.org/www-community/initiatives/gsoc/gsoc2022ideas
  - https://owasp.org/www-community/initiatives/gsoc/gsoc2023ideas
to gather participating project names and put them into JSON files the same way it was done [here](https://github.com/OWASP/Nest/commit/d26217231743a16b3cae2140738b12c1ce232660)

After it's done the `backend/data/project-custom-tags` should include gsoc-20[20-24].json files

### PR DESCRIPTION
**Pull request by nitinawari**
close #167 
- Added JSON files for GSoC project custom tags for the years 2020–2023.
- Files are located in `backend/data/project-custom-tags/`.

### CODE REVIEW

File: backend/data/project-custom-tags/gsoc-2020.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "projects": [
+    "www-project-python-honeypot", 
+    "www-project-zed-attack-Proxy",
```
Comment (arkid15r) [inline backend/data/project-custom-tags/gsoc-2020.json]: Could you make sure the keys are the same that we have for the projects in our DB, e.g. `www-project-zed-attack-Proxy` should be `www-project-zap` (the same names projects have in the OWASP GitHub organization repositories -- https://github.com/orgs/OWASP/repositories)

Comment (nitinawari) [reply backend/data/project-custom-tags/gsoc-2020.json]: "I’ve cross-checked the project search with both **Algolia** and the **Nest website** and **owasp repo**, and all keys are now searchable.

 For **ZAP**, there is no repo in the database, so I added the full project name earlier. **python-honeypot**, **ZAP**, and **security-knowledge-framework** were not found in the database, which is why I asked for clarification on how to handle those projects. 
Apologies for the earlier precommit issue; I’ve made the necessary adjustments now."

Comment (nitinawari) [reply backend/data/project-custom-tags/gsoc-2020.json]: Let me know if there’s anything that needs to be updated!

Comment (arkid15r) [reply backend/data/project-custom-tags/gsoc-2020.json]: What database do you refer to? 
You can check all existing projects via Django admin of your local setup (the source file for all data is `backend/data/nest.json.gz`

**Comment by yashpandey06** [comment]
@nitinawari  you can run it via Makerfile  using command make  check .

File: backend/data/project-custom-tags/gsoc-2020.json (hunk 1 above)
Comment (nitinawari) [reply backend/data/project-custom-tags/gsoc-2020.json]: "I’m still learning Django and how databases are handled. The local setup is a bit messy, so I’ve only created the JSON files. Let me know if I missed any step or if there’s an error. Also, should I update ZAP, Python Honeypot, and SFK in the database as part of this task, or is there something else I need to do?

**Comment by nitinawari** [comment]
@yashpandey06  i run make precommit

**Review by arkid15r — APPROVED** [review]
Thanks for doing this chore for us!
I've updated some keys and merging this PR now.

**Pull request merged**